### ISSUE TITLE
Some differences from the original code

### ISSUE THREAD

**Issue by zeal-up** [hxdengBerkeley/PointCNN.Pytorch#5]
Thanks so much for your works! I try to reproduce the results of PointCNN in Pytorch but still cannot get a reasonable result and I  find your project just now. But I have some question about the code you provided.
The network in this repository is some kind of different from the origin one.
1, the author uses three depthwise conv layer to produce the transform matrix X. You just use one depthwise layer plus another two dense layer. It will need more parameters compared to the original and maybe arise overfitting.
2, why do you remove batchnorm in your dense layer? I think this is unreasonable.
3, the configuration is quite different from the original one
![image](https://user-images.githubusercontent.com/40826585/49157014-c2c30600-f359-11e8-9a41-9c7aedc728e0.png)
in the original code, the output channel has a coefficient 3. And it seems that you use 5 xconv layers while the author uses 4.
4, what's the best accuracy you get on ModelNet40?
Nevertheless, thanks for your sharing. Hope my questions are not too much.

**Comment by leafxx** [comment]
> Thanks so much for your works! I try to reproduce the results of PointCNN in Pytorch but still cannot get a reasonable result and I find your project just now. But I have some question about the code you provided.
> The network in this repository is some kind of different from the origin one.
> 1, the author uses three depthwise conv layer to produce the transform matrix X. You just use one depthwise layer plus another two dense layer. It will need more parameters compared to the original and maybe arise overfitting.
> 2, why do you remove batchnorm in your dense layer? I think this is unreasonable.
> 3, the configuration is quite different from the original one
> ![image](https://user-images.githubusercontent.com/40826585/49157014-c2c30600-f359-11e8-9a41-9c7aedc728e0.png)
> in the original code, the output channel has a coefficient 3. And it seems that you use 5 xconv layers while the author uses 4.
> 4, what's the best accuracy you get on ModelNet40?
> Nevertheless, thanks for your sharing. Hope my questions are not too much.

hi, what accracy you get by using this repo?

**Comment by tengteng95** [comment]
I also wonder the accuracy achieved by this repo. I tried but cannot get a reasonable results.

**Comment by zeal-up** [comment]
@jw-j I have not train the model in this repo because it's different from the original paper. I just wander how the author of this repo construct this model.

**Comment by Jimmy880** [comment]
@zeal-github I have trained this model and i found that the loss can not even decrease after several epoch on the modelnet dataset. Now, i am doubt about the correctness of this repo.

**Comment by htkang369** [comment]
I met the same problem. After train even 7 epoches, validation accuracy is the same as previous ones, didn't change at all...

**Comment by LiuNull** [comment]
> @jw-j I have not train the model in this repo because it's different from the original paper. I just wander how the author of this repo construct this model.

Hi, have you found a correct pointcnn pytorch project which could match the accuracy of the paper?

**Comment by jxqhhh** [comment]
> @zeal-github I have trained this model and i found that the loss can not even decrease after several epoch on the modelnet dataset. Now, i am doubt about the correctness of this repo.

Hi, have you recorded the best accuracy of this project? I'm trying to implement the model with Pytorch, but I just achieve an accuracy of 20%

**Comment by jxqhhh** [comment]
> > @jw-j I have not train the model in this repo because it's different from the original paper. I just wander how the author of this repo construct this model.
> 
> Hi, have you found a correct pointcnn pytorch project which could match the accuracy of the paper?

I'm implementing it right now. I will make my implementation public in a few days. Maybe you can choose to follow me in Github😂

**Comment by anewlearner** [comment]
> > > @jw-j I have not train the model in this repo because it's different from the original paper. I just wander how the author of this repo construct this model.
> > 
> > 
> > Hi, have you found a correct pointcnn pytorch project which could match the accuracy of the paper?
> 
> I'm implementing it right now. I will make my implementation public in a few days. Maybe you can choose to follow me in Github😂

Hi jxqhhh,
Have you finished yet? Look forward to it.

**Comment by jxqhhh** [comment]
> Look
I will post my code a few hours later here:
https://github.com/jxqhhh/PytorchPointCNN

**Comment by jxqhhh** [comment]
> > > > @jw-j I have not train the model in this repo because it's different from the original paper. I just wander how the author of this repo construct this model.
> > > 
> > > 
> > > Hi, have you found a correct pointcnn pytorch project which could match the accuracy of the paper?
> > 
> > 
> > I'm implementing it right now. I will make my implementation public in a few days. Maybe you can choose to follow me in Github😂
> 
> Hi jxqhhh,
> Have you finished yet? Look forward to it.

I will post my code a few hours later here:
https://github.com/jxqhhh/PytorchPointCNN

**Comment by jxqhhh** [comment]
> I also wonder the accuracy achieved by this repo. I tried but cannot get a reasonable results.

try this repository:
https://github.com/jxqhhh/PytorchPointCNN

**Comment by LiDaiY** [comment]
I slightly modified the code and trained it for classification task，and it seems to encounter the problem of overfitting. At the 38th epoch, the Train Instance Accuracy: 0.802424, while the test only is 0.637903. And the Best Instance Accuracy: 0.796774 occured in epoch 36.

**Comment by LiDaiY** [comment]
> I slightly modified the code and trained it for classification task，and it seems to encounter the problem of overfitting. At the 38th epoch, the Train Instance Accuracy: 0.802424, while the test only is 0.637903. And the Best Instance Accuracy: 0.796774 occured in epoch 36.

It might be the problem of dataset, I replaced the data with H5 type that processed by pointnet, and the training seems going well:
Epoch 38 (38/200):                                                                                                                                    Train Instance Accuracy: 0.756849                                                                                                                    Test Instance Accuracy: 0.837925, Class Accuracy: 0.786131                                                                                           Best Instance Accuracy: 0.846434, Class Accuracy: 0.787909
